components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java

```diff
@@ -200,6 +211,35 @@ public void invoke(Request request, Response response) throws IOException, Servl
 
     }
 
+    /**
+     * This method is used to override the secured resource based on tenant-wise DCR api security configuration.
+     * @param securedResource securedResource object
+     * @param normalizedRequestURI request URL path
+     * @param httpMethod http method
+     * @param tenantDomain tenant domain of the request
+     * @throws DCRMException DCRMException
+     */
+    private void overrideSecuredResource(ResourceConfig securedResource, String normalizedRequestURI, String httpMethod,
+                                        String tenantDomain) throws DCRMException {
+
+        if (normalizedRequestURI.contains(DCR_REGISTER_ENDPOINT_PATH) && HttpMethod.POST.equals(httpMethod)) {
+
+            if (DCRMgtOGSiServiceFactory.getInstance() != null) {
+
+                DCRConfiguration dcrConfiguration = DCRMgtOGSiServiceFactory.getInstance().getDCRConfiguration();
+                Boolean isClientAuthenticationRequired = dcrConfiguration.isAuthenticationRequired();
+                if ((Boolean.TRUE).equals(isClientAuthenticationRequired)) {
+                    securedResource.setIsSecured(true);
+                } else if ((Boolean.FALSE).equals(isClientAuthenticationRequired)) {
```

Comment:
Shouldn't we need to check else clause also where isClientAuthenticationRequired=null

Reply:
`isClientAuthenticationRequired` being `null` acts as the current existing behaviour and I haven't changed that. By default `securedResource.isSecured` is true unless `isClientAuthenticationRequired` is not specified.